app/src/main/java/com/android/feedme/MainActivity.kt

```diff
@@ -59,11 +62,16 @@ class MainActivity : ComponentActivity() {
                 }
 
                 navigation(startDestination = Screen.HOME, route = Route.HOME) {
-                  composable(Screen.HOME) { LandingPage(navigationActions) }
-                  composable(Screen.RECIPE) { backStackEntry ->
-                    backStackEntry.arguments?.getString("recipeId")?.let {
-                      RecipeFullDisplay(navigationActions)
-                    }
+                  composable(Screen.HOME) {
+                    // Create a shared view model for Recipe
+                    val recipeViewModel = viewModel<RecipeViewModel>()
+                    LandingPage(navigationActions, recipeViewModel)
+                  }
+                  composable(Screen.RECIPE) {
+                    // Link the shared view model to the composable
+                    val navBackStackEntry = navController.getBackStackEntry(Screen.HOME)
+                    val recipeViewModel = viewModel<RecipeViewModel>(navBackStackEntry)
```

Comment:
Another problem may be, to be sure that the `RecipeViewModel` is set to the correct recipe we want.
One fix for that would be to kept the Id navigation and compare it upon creation to fetch the `Recipe` if needed.
However, this scenario should never happen as we always should be able to set the `Recipe` we want to display before hand. Do you agree ?

Reply:
We are sure that we'll have the correct recipe. When navigating to `RecipeFullDisplay` from `LandingPage`, I'm selecting the `Recipe` we click on before navigating to the screen using `selectRecipe(recipe)` on the shared `ViewModel` (on line 132 of `LandingScreen.kt`). 
Does this answer your doubt ?